transformers/struct.go

```diff
@@ -159,6 +256,11 @@ func (t *structTransformer) addColumnFromField(field reflect.StructField, parent
 		IgnoreInTests: t.ignoreInTestsTransformer(field),
 	}
 
+	// Avoid infinite recursion
+	if columnType == types.ExtensionTypes.JSON && t.jsonTypeSchemaDepth < maxJSONTypeSchemaDepth {
```

Comment:
The diff is ugly but `addColumnFromField` didn't change much, only extracted `getColumnType` and added the condition with a call to `transformFieldToSchema` that handles getting the schema for the JSON type